ironfish/src/rpc/routes/chain/getBlocks.ts

```diff
@@ -0,0 +1,100 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import * as yup from 'yup'
+import { Assert } from '../../../assert'
+import { getBlockSize } from '../../../network/utils/serializers'
+import { FullNode } from '../../../node'
+import { GENESIS_BLOCK_SEQUENCE } from '../../../primitives/block'
+import { RpcNotFoundError, RpcValidationError } from '../../adapters'
+import { ApiNamespace } from '../namespaces'
+import { routes } from '../router'
+import { RpcBlock, RpcBlockSchema, serializeRpcBlockHeader } from '../types'
+import { RpcTransaction } from './types'
+import { serializeRpcTransaction } from './utils'
+
+const MAX_BLOCKS_RANGE = 30
+
+export type GetBlocksRequest = {
+  /**
+   * The starting block height (inclusive).
+   */
+  start: number
+  /**
+   * The ending block height (exclusive).
+   */
+  end: number
+}
+
+export type GetBlocksResponse = {
+  blocks: RpcBlock[]
+}
+
+export const GetBlocksRequestSchema: yup.ObjectSchema<GetBlocksRequest> = yup
+  .object()
+  .shape({
+    start: yup.number(),
+    end: yup.number(),
+  })
+  .defined()
+
+export const GetBlocksResponseSchema: yup.ObjectSchema<GetBlocksResponse> = yup
+  .object({
+    blocks: yup.array(RpcBlockSchema).defined(),
+  })
+  .defined()
+
+routes.register<typeof GetBlocksRequestSchema, GetBlocksResponse>(
+  `${ApiNamespace.chain}/getBlocks`,
+  GetBlocksRequestSchema,
+  async (request, context): Promise<void> => {
+    Assert.isInstanceOf(context, FullNode)
+
+    // Use negative numbers to start from the genesis block of the chain
+    if (request.data.start <= 0) {
+      request.data.start = Math.max(request.data.start, GENESIS_BLOCK_SEQUENCE)
+    }
+
+    if (request.data.end - request.data.start > MAX_BLOCKS_RANGE) {
+      request.data.end = request.data.start + MAX_BLOCKS_RANGE
+    }
+
+    const blocks: RpcBlock[] = []
+    for (let seq = request.data.start; seq < request.data.end; seq++) {
+      const block = await getBlockWithSequence(context, seq)
+      blocks.push(block)
+    }
+
+    request.end({ blocks })
+  },
+)
+
+const getBlockWithSequence = async (node: FullNode, sequence: number): Promise<RpcBlock> => {
```

Comment:
nit: we use the word `getBlockAtSequence` like the function you called on line 73 here https://github.com/iron-fish/ironfish/blob/8b26bfead96c577399e4cccd5879dcb785194307/ironfish/src/blockchain/blockchain.ts#L1062

So this should be named `getBlockAtSequence`